### PR TITLE
fix: Errors shortly after SentrySDK.init don't affect the session

### LINKED ISSUE #2150: Errors shortly after SentrySDK.init don't affect the session
### Description

Capturing an error direclty after calling SentrySDK.init doesn't affect the session error count, as we start the session when `didBecomeActive` is called.

https://github.com/getsentry/sentry-cocoa/blob/f273eda1e1d73faa43e66fa4330e336347ab46ec/Sources/Sentry/SentrySessionTracker.m#L131-L163

We could start the session during the SDK init / when initializing the SessionIntegration.

Related to https://github.com/getsentry/sentry-java/issues/2241.

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

Errors shortly after SentrySDK.init now do affect the session.

## :bulb: Motivation and Context

Closes #2150

## :green_heart: How did you test it?

Unit test

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/include/SentrySession+Private.h [hunk 1]
```diff
@@ -8,6 +8,8 @@ NSString *nameForSentrySessionStatus(SentrySessionStatus status);
 @interface
 SentrySession (Private)
```
Comment (armcknight) [inline Sources/Sentry/include/SentrySession+Private.h:9]: I made this refactoring elsewhere: change this from a category to a class extension (anonymous category) then you don't have to redeclare the property in SentrySession.m; instead, you can just `#include "SentrySession+Private.h"

**Review by armcknight — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ This version adds a dependency on Swift.
 
 - Properly demangle Swift class name (#2162)
 
+### Fixes
+
+- Errors shortly after SentrySDK.init now affect the session (#2430)
```
Comment (philipphofmann) [inline CHANGELOG.md:13]: Excellent changelog entry 👏

File: Tests/SentryTests/SentryHubTests.swift [hunk 3]
```diff
@@ -394,6 +394,19 @@ class SentryHubTests: XCTestCase {
         XCTAssertEqual(1, fixture.client.captureSessionInvocations.count)
     }
 
+    func testCaptureErrorBeforeSession() {
```
Comment (philipphofmann) [inline Tests/SentryTests/SentryHubTests.swift:397]: `h`: We must set the `_errorsBeforeSession` to `0` in `SentryHub.endSession`. Users could disable auto session tracking and manually track sessions. Please also add a test for this.

File: Sources/Sentry/SentryHub.m [hunk 4]
```diff
@@ -93,6 +96,11 @@ - (void)startSession
         }
         _session = [[SentrySession alloc] initWithReleaseName:options.releaseName];
 
+        if (_errorsBeforeSession > 0) {
```
Comment (philipphofmann) [inline Sources/Sentry/SentryHub.m:99]: `h`: We should only do this if `options.enableAutoSessionTracking == true`. We would mess up the statistics if the user switches to manual mode.

Comment (kevinrenskers) [reply Sources/Sentry/SentryHub.m]: Seems a bit weird to me. The user captured an error, and they started a session.. so no matter if `enableAutoSessionTracking` is on or off, that session's error count should be increased I would think?

But I made the change and added a test for it.

**Review by philipphofmann — APPROVED** [review]
Thanks

**Pull request merged**